### PR TITLE
js[patch]: Fix listRuns limit arg

### PR DESCRIPTION
**Pull request by bracesproul**
Added faker package for tests

### CODE REVIEW

**Review by hinthornw — APPROVED** [review]

File: js/src/client.ts [hunk 1]
```diff
@@ -1194,11 +1194,25 @@ export class Client {
...
       "/runs/query",
       body
     )) {
-      yield* runs;
+      if (limit) {
+        if (runsYielded >= limit) {
+          break;
+        }
+        if (runs.length + runsYielded > limit) {
+          const newRuns = runs.splice(limit - runsYielded);
```
Comment (dqbd) [inline js/src/client.ts:1207]: Shouldn't it be 

```suggestion
          const newRuns = runs.slice(0, limit - runsYielded)
```

Comment (dqbd) [reply js/src/client.ts:1207]: Or alternatively

```suggestion
          runs.splice(limit - runsYielded);
          yield* runs
```

**Comment by bracesproul** [comment]
@hinthornw ci throwing 401 unauthorized errors. Did something happen to the LS secrets in gh actions?

**Comment by hinthornw** [comment]
Nothing was changed in 3 months... Wonder if GH is acting up. I went ahead and updated the key just in case

**Pull request merged**